### ISSUE TITLE
gdi01/gdi02: Lua exception "LoadPassenger requires the passenger to be idle" at mission start

### ISSUE THREAD

**Issue by shihan42** [OpenRA/OpenRA#21156]
## Issue Summary

At the start of gdi01 and after a few seconds in gdi02, a lua exception is thrown.
"LoadPassenger requires the passenger to be idle."

https://github.com/OpenRA/OpenRA/blob/a3c0cee2ccc43d7d04e4a48e91af249be6b2ecdd/OpenRA.Mods.Common/Scripting/Properties/TransportProperties.cs#L45

It is called through

https://github.com/OpenRA/OpenRA/blob/a3c0cee2ccc43d7d04e4a48e91af249be6b2ecdd/mods/cnc/scripts/campaign.lua#L36C9

The actor in question is the MCV. I stepped through with the debugger and found the CurrentActivity to be `OpenRA.Mods.Common.Traits.Mobile.LeaveProductionActivity`. That leads to `IsIdle == false`, which triggers the exception.

I tried to edit the script and explicitly stop the activity, but it didn't work.

```lua
ReinforceWithLandingCraft = function(player, units, transportStart, transportUnload, rallypoint)
	local transport = Actor.Create("lst", true, { Owner = player, Facing = Angle.North, Location = transportStart })
	local subcell = 0
	Utils.Do(units, function(a)
		local newUnit = Actor.Create(a, false, { Owner = transport.Owner, Facing = transport.Facing, Location = transportUnload, SubCell = subcell })
		-- immediately stop the actor
		newUnit.Stop()
		-- throws regardless
		transport.LoadPassenger(newUnit)
		subcell = subcell + 1
	end)
```

Disclaimer: I'm using the Gold Edition ISOs.

## Reproduction

1. Compile a3c0cee2ccc43d7d04e4a48e91af249be6b2ecdd
2. Start mod `cnc`
3. Start singleplayer missions gdi01 or gdi02

## Expected behavior

1. `ReinforceWithLandingCraft` creates units without a current activity.
2. The lua exception does not rise.

**Comment by PunkPun** [comment]
Thanks for the report. The issue is known and already has a fixed lined up #21076

**Comment by shihan42** [comment]
Ah, that's great :)

Sorry that I didn't search for existing pull requests, only for existing issues.
